### PR TITLE
Y24-238 - Update SMRT Link Versioning documentation

### PR DESCRIPTION
**Pull request by stevieing**
Closes #

#### Changes proposed in this pull request

- Created SMRT Link Versioning documentation
- Fixed Yard warnings
- Modified some of the internal docs to better reflect usage

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: documentation/smrt-link-versioning/docs/creating-a-new-smrt-link-version.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Creating a new SMRT Link Version
+
+1. Create a new SMRT Link Version in the configuration file [`config/pacbio_smrt_link_versions.yml`](https://github.com/sanger/traction-serice/blob/develop/config/pacbio_smrt_link_versions.yml):
```
Comment (dasunpubudumal) [inline documentation/smrt-link-versioning/docs/creating-a-new-smrt-link-version.md:3]:
````suggestion
1. Create a new SMRT Link Version in the configuration file [`config/pacbio_smrt_link_versions.yml`](https://github.com/sanger/traction-service/blob/61e0fff5de4f7ad83cf579dae27bfdea020f0d48/config/pacbio_smrt_link_versions.yml):
```

Comment (dasunpubudumal) [inline documentation/smrt-link-versioning/docs/creating-a-new-smrt-link-version.md:3]:
```suggestion
1. Create a new SMRT Link Version in the configuration file [`config/pacbio_smrt_link_versions.yml`](https://github.com/sanger/traction-service/blob/61e0fff5de4f7ad83cf579dae27bfdea020f0d48/config/pacbio_smrt_link_versions.yml):
```

File: documentation/smrt-link-versioning/docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    - This configuration provides details of instrument type structure and validation.
+
+    - For example for Revio the structure is run > plate > wells:
+      - A Revio run can have 1 or 2 plates
+      - A revio plate must have a sequencing kit box barcode
+      - Revio wells can appear in a number of specific orders
+
+  2. Pacbio - [`config/pipelines/pacbio.yml`](https://github.com/sanger/traction-service/blob/develop/config/pipelines/pacbio.yml) - contains details on the sample sheet. When a sample sheet is created this file is used to drive the attributes that will appear, where that data comes from and any class that is used to drive behaviour
+
+  3. Pacbio SMRT Link Versions - [`config/pacbio_smrt_link_versions.yml`](https://github.com/sanger/traction-service/blob/develop/config/pacbio_smrt_link_versions.yml) - This file is loaded into the database using a rake task. This data is then used in run creation and validation.
```
Comment (seenanair) [inline documentation/smrt-link-versioning/docs/architecture.md:92]: Would it be helpful to specify which rake task it is

**Review by seenanair — COMMENTED** [review]
Looking great. 'Creating a new SMRT Link Version' is really useful for future changes. 👍 

One minor thing, all other documentation pages are displayed with a traction icon which is missing in  smrt-link-versioning page. (I think  favicon.ico need to be added in documentation/smrt-link-versioning/docs/img folder)

**Review by dasunpubudumal — APPROVED** [review]

**Review by seenanair — APPROVED** [review]

**Pull request merged**